tidy3d/components/mode/mode_solver.py

```diff
@@ -498,6 +546,29 @@ def _get_data_with_group_index(self) -> ModeSolverData:
 
         return mode_solver.data_raw._group_index_post_process(self.mode_spec.group_index_step)
 
+    def _get_data_with_interp(self) -> ModeSolverData:
+        """:class:`.ModeSolverData` computed at reduced frequencies and interpolated back.
+
+        Returns
+        -------
+        ModeSolverData
+            :class:`.ModeSolverData` object with modes computed at a reduced set of frequencies
+            specified by ``interp_spec.num_points`` and interpolated to the original frequencies.
+        """
+
+        # Create reduced frequency set based on interpolation method
+        freqs_reduced = self.interp_spec.sampling_points(self.freqs)
+
+        # Create a copy of the mode solver with reduced frequencies and no interp_spec
+        # (to prevent recursion)
+        mode_solver_reduced = self.copy(update={"freqs": freqs_reduced, "interp_spec": None})
+
+        # Get data at reduced frequencies
+        data_reduced = mode_solver_reduced.data_raw
```

Comment:
Can you use `_data_on_yee_grid` here instead, and then do the processing later in `data_raw`?